exporter/googleservicecontrolexporter/logs.go

```diff
@@ -0,0 +1,493 @@
+		// Keep start_time = now - 1ms.
+		StartTime:   timestamppb.New(now.Add(-1 * time.Second)),
+		EndTime:     timestamppb.New(now),
+		OperationId: uuid.New(),
+		Labels:      mr,
+		LogEntries:  le,
+	}
+	return &op, nil
+}
+
+func (e *LogsExporter) createEntries(rl plog.ResourceLogs) ([]*scpb.LogEntry, map[string]string, string, error) {
+	var errs []error
+	entries := make([]*scpb.LogEntry, 0)
+	processTime := time.Now()
+	resourceAttributes, consumerId := e.parseResourceAttributes(rl.Resource())
+	for j := range rl.ScopeLogs().Len() {
+		sl := rl.ScopeLogs().At(j)
+		// TODO: handle otel instrumentation scope labels, i.e., instrumentation_source
+		// and instrumentation_version
+		for k := range sl.LogRecords().Len() {
+			logRecord := sl.LogRecords().At(k)
+			entry, err := e.logMapper.parseLogEntry(logRecord, processTime)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			entries = append(entries, entry)
+		}
+	}
+
+	return entries, resourceAttributes, consumerId, errors.Join(errs...)
+}
+
+func (l logMapper) getLogName(log plog.LogRecord) (string, error) {
+	logNameAttr, exists := log.Attributes().Get(LogNameAttributeKey)
+	if exists {
+		return logNameAttr.AsString(), nil
+	}
+	if len(l.cfg.LogConfig.DefaultLogName) > 0 {
+		return l.cfg.LogConfig.DefaultLogName, nil
+	}
+	return "", fmt.Errorf("encountered log without 'log_name' field, while 'default_log_name' value in configuration is empty")
+}
+
+// parseLogEntry creates a Service Control LogEntry from otel LogRecord. Service
+// Control API does not support log splits, so one otel log will always be one
+// log entry.
+func (l logMapper) parseLogEntry(logRecord plog.LogRecord, processTime time.Time) (*scpb.LogEntry, error) {
+	ts := logRecord.Timestamp().AsTime()
+	if logRecord.Timestamp() == 0 || ts.IsZero() {
+		// if timestamp is unset, fall back to observed_time_unix_nano as recommended
+		//   (see https://github.com/open-telemetry/opentelemetry-proto/blob/4abbb78/opentelemetry/proto/logs/v1/logs.proto#L176-L179)
+		if logRecord.ObservedTimestamp() != 0 {
+			ts = logRecord.ObservedTimestamp().AsTime()
+		} else {
+			// if observed_time is 0, use the process time, which is the current time
+			ts = processTime
+		}
+	}
+
+	logName, err := l.getLogName(logRecord)
+	if err != nil {
+		return nil, err
+	}
+
+	entry := &scpb.LogEntry{
+		Name:      logName,
+		Timestamp: timestamppb.New(ts),
+		Labels:    map[string]string{},
+	}
+
+	// build our own map off OTel attributes so we don't have to call .Get() for each special case
+	// (.Get() ranges over all attributes each time)
+	attrsMap := make(map[string]pcommon.Value)
+	logRecord.Attributes().Range(func(k string, v pcommon.Value) bool {
+		attrsMap[k] = v
+		return true
+	})
+
+	// Parse LogEntry InsertId struct from OTel attribute
+	// TODO: we should evaluate if this is needed; if not, can remove this and
+	// let the API generate a UUID - same behavior as the Cloud Logging exporter
```

Comment:
What did the UCP logging agent do to generate the insertId? The comment below suggests that the API would handle it, maybe we should do the same

Reply:
Updated the comment. 
For UCP agent: first parse from the attributes, and if not present, generate a UUID in the exporter. 
For Cloud Logging exporter: do not handle - let API generate UUID. 

Here I kept parsing from the attributes, but removed the UUID generating (helps simplify testing).